design/pool-configuration.md

```diff
@@ -0,0 +1,390 @@
+# Address Pool configuration options
+
+## Summary
+
+Following @johananl's initiative in [this issue](https://github.com/metallb/metallb/issues/630)
+this enhancement tries to collect and classify various configuration enhancement requests
+requested over time.
+
+## Motivation
+
+The way advanced users consume MetalLB is evolving and exposed a need for more complex configuration.
+For example, the need for isolation and multi-tenancy induced requests for being able to advertise some specific address pools from a subset of the nodes, or
+to allocate them only to a specific namespace, to mention some.
+
+### Goals
+
+* Collect and classify all the requests coming from different issues, in order to have a better understanding
+  and an organic implementation, avoiding to focus on and to implement only a specific feature, risking to cause conflicts afterwards.
+* Provide a list of the possible configuration enhancements, and to highlight dependencies and conflicts between each other (if any).
+* Highlight the cases where the solution must be BGP or L2 specific
+
+### Non-Goals
+
+* Covering only a specific use case.
+* Providing a L2 only or BGP only solution where the problem might be expressed in a more generic way.
+
+## Proposal
+
+For all the ips belonging to an address pool, we need to address three main points:
+
+* Which nodes should advertise the ip
+* To which BGP neighbors the IP address should be advertised
+* How to allocate that ip to a given service
+
+The general idea of this proposal is that these degrees of configuration are all related to the AddressPool object, and
+that these configurations are independent between each other.
+
+### Node / Address Pool mapping
+
+The most common request is being able to associate a given pool to a subset of nodes (which translates to selectively announce a subset of speakers to announce the ip from).
```

Comment:
```suggestion
The most common request is being able to associate a given pool to a subset of nodes (which translates to selecting a subset of speakers so that these speakers - and only them - announce the IP).
```

I think there is a phrasing problem here.